vaas/vaas/cluster/mapping.py

```diff
@@ -10,11 +10,18 @@ def __init__(self, mappings: List[DomainMapping]):
             'dynamic': [],
         }
         for m in mappings:
-            self.mappings[m.type].append(m)
+            self.mappings[m.type].append(
+                {
+                    'cluster_keys': [c.pk for c in m.clusters.all()],
+                    'mapping': m
+                }
+            )
 
     def provide_related_domains(self, cluster: LogicalCluster) -> List[str]:
-        result = {m.domain for m in cluster.domainmapping_set.filter(type='static')}
+        print(cluster.name)
+        result = {m['mapping'].domain for m in self.mappings['static'] if cluster.pk in m['cluster_keys']}
+        cluster_labels = set(list(cluster.parsed_labels().keys()))
```

Comment:
I suppose the keys() will return dict_keys(), which are unique so there is no need to make a list of them or even set. But we are using a difference function so only set could be left here.